code/modules/client/preferences2/character_save.dm

```diff
@@ -469,6 +469,7 @@
 		pref_species = new /datum/species/human
 		save(usr.client, async = FALSE) // This entire proc is called a lot at roundstart, and we dont want to lag that
 
+	character.dna.features = features.Copy()
 	character.set_species(chosen_species, icon_update = FALSE, pref_load = TRUE)
 	character.dna.features = features.Copy()
```

Comment:
This cannot be the best way to handle this. What _actually_ causes all this stuff to break?

Reply:
![image](https://user-images.githubusercontent.com/9547572/215687492-b888a7d4-7186-4aec-8dfb-988b199ecbb1.png)

Reply:
Long story short, without 
`character.dna.features = features.Copy()` before
`character.set_species(chosen_species, icon_update = FALSE, pref_load = TRUE)`
When `character.set_species(chosen_species, icon_update = FALSE, pref_load = TRUE)` will not have features updated to spawn the person in with digi legs.

Now, as to why `character.dna.features = features.Copy()` after
`character.set_species(chosen_species, icon_update = FALSE, pref_load = TRUE)` fixes the preview screen, I do not know. The update occurs somewhere within /client/proc/send_resources(), most notably, around line 964. I cannot drill down any further than this.
 I do not know why this update doesn't happen without `character.dna.features = features.Copy()` after `character.set_species(chosen_species, icon_update = FALSE, pref_load = TRUE)`.